checks/ulimits.go

```diff
@@ -0,0 +1,121 @@
+package checks
+
+import (
+	"fmt"
+	"syscall"
+)
+
+// UlimitChecker checks if current process resource limits are above a given minimum
+//
+// Type:
+//   - ulimit
+//
+// Supported Platforms:
+//   - MacOS
+//   - Linux
+//
+// Arguments:
+//   - item (required): A string value representing the type of limit to check
+//   - limit (required): Numerical value representing the minimum value to be tested
+//   - type (optional): "hard" or "soft" with a default of "hard"
+//
+// Notes:
+//   - Windows is not supported
```

Comment:
Seems redundant with the support platforms above

Reply:
Also if this is not for Windows it should use a build tag (preferrably file name) to only build for unix

Reply:
removed that note line and moved files to ulimits_unix.go and ulimits_unix_test.go